contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -2540,6 +2540,119 @@ CREATE OR REPLACE FUNCTION sys.json_query(json_string text, path text default '$
+    json_path TEXT;
+    json_path_convert TEXT;
+    new_jsonb_path TEXT[];
+    type_key_value TEXT;
+    mode TEXT;
+    modifier TEXT;
+    len_array INTEGER;
+    word_count INTEGER;
+    create_if_missing BOOL:='TRUE';
+    append_modifier BOOL:='FALSE';
+    key_exist BOOL;
+    key_value JSONB;
+    json_expression JSONB;
+    result_json NVARCHAR;
+BEGIN
+    json_expression = to_jsonb(expression::JSONB);
+
+    json_path = regexp_replace(path_json, '^.* ', '','ig'); -- To select the last word of the json_path string. \\\\\\Instead of this one can use split_part if regexp is slow
+    json_path_convert = regexp_replace(json_path, '\$\.|]|\$\[' , '' , 'ig'); -- To remove the "$." and "]" sign from the string 
+    json_path_convert = regexp_replace(json_path_convert, '\.|\[' , ',' , 'ig'); -- To replace the "." and "[" with the "," to change into required format
+    new_jsonb_path = CONCAT('{',json_path_convert,'}'); -- Final required format of path by jsonb_set
+    
+    
+    key_exist = jsonb_path_exists(json_expression,json_path::jsonpath); -- TO check if key exist in the given path
+    word_count = LENGTH(path_json) - LENGTH(REPLACE(path_json, ' ', '')) + 1; --To calculate the number of words in the path_json
+    
+
+    -- This if else block is added to set the create_if_missing and append_modifier flags
+    IF word_count=1 THEN --The path_json has only 1 word
+        create_if_missing='TRUE';
+        append_modifier='FALSE';
+    ELSIF word_count=2 THEN --The path_json has 2 words
+        mode = SPLIT_PART(path_json COLLATE sql_latin1_general_cp1_cs_as, ' ',1);
+        IF mode='append' THEN
+            append_modifier:='TRUE';
+        ELSIF mode='strict' THEN
+            create_if_missing:='FALSE';
+        END IF;
+    ELSIF word_count=3 THEN --The path_json has 3 words
+        modifier = SPLIT_PART(path_json COLLATE sql_latin1_general_cp1_cs_as, ' ',1);
+        mode = SPLIT_PART(path_json COLLATE sql_latin1_general_cp1_cs_as, ' ',2);
+        IF modifier='append' THEN
+            append_modifier:='TRUE';
+        END IF;
+        IF mode='strict' THEN
+            create_if_missing:='FALSE';
+        END IF;
+    END IF;
+
+
+    --This if else block is to call the jsonb_set function based on the create_if_missing and append_modifier flags
+    
+    IF append_modifier THEN --When append modifier is present
+        IF key_exist THEN
+            key_value = jsonb_path_query_first(json_expression,json_path::jsonpath); -- To get the value of the key
+            type_key_value = jsonb_typeof(key_value);
+            IF type_key_value='array' THEN
+                len_array = jsonb_array_length(key_value);
+                new_jsonb_path = CONCAT(TRIM('}' FROM new_jsonb_path::TEXT),',',CAST(len_array AS TEXT),'}'); -- To change the path format into the required jsonb_insert path format
+                IF new_value IS NULL THEN
+                    result_json = jsonb_insert(json_expression,new_jsonb_path,'null');
```

Comment:
Please add a comment that we can't simply do "to_jsonb(coalesce(new_value, 'null'))" here, because that won't result in a JSON NULL".